src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,148 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.nextElement = new Node<>(tail, value, null);
+            tail = tail.nextElement;
```

Comment:
in our variants you will be initialize a new Node<>(tail, value, null);
so i think you can do it at start at the method and later just change the values) if you agree of course)

Reply:
these nodes have different parameters